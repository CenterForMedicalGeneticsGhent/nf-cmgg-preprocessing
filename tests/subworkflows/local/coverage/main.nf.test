nextflow_workflow {

    name "Test Workflow COVERAGE"
    script "subworkflows/local/coverage/main.nf"
    workflow "COVERAGE"

    test("coverage targetted bam") {

        when {
            params {
                run_coverage = true
                disable_picard_qc = false
            }
            workflow {
                """
                // ch_bam_bai [meta, bam, bai]
                input[0] = Channel.of([
                    [id: 'test', single_end: false],
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                ])

                // ch_fasta_fai [meta2, fasta, fai]
                input[1] = Channel.of([
                    [id: 'sarscov2'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true),
                ])

                // ch_fasta_dict [meta2, dict]
                input[2] = Channel.of([
                    [id: 'sarscov2'],
                    file(params.test_data['sarscov2']['genome']['genome_dict'], checkIfExists: true),
                ])

                // Target interval file
                input[3] = Channel.of(file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)),

                // Bait interval file
                input[4] = Channel.of(file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true))
                """
            }
        }

        then {
            assert workflow.success
            // check number of processes executed
            assert workflow.trace.succeeded().size() == 2
        }

    }

    test("coverage wgs bam") {

        when {
            params {
                run_coverage = true
                disable_picard_qc = false
            }
            workflow {
                """
                // ch_bam_bai [meta, bam, bai]
                input[0] = Channel.of([
                    [id: 'test', single_end: false],
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                    file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                ])

                // ch_fasta_fai [meta2, fasta, fai]
                input[1] = Channel.of([
                    [id: 'sarscov2'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true),
                ])

                // ch_fasta_dict
                input[2] = Channel.empty()

                // Target interval file
                input[3] = Channel.empty()

                // Bait interval file
                input[4] = Channel.empty()
                """
            }
        }

        then {
            assert workflow.success
            // check number of processes executed
            assert workflow.trace.succeeded().size() == 2
        }

    }

}
