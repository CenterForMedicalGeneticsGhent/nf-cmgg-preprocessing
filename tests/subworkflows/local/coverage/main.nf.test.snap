{
    "COVERAGE - Target interval file": {
        "content": [
            {
                "per_base_bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.per-base.bed.gz:md5,3bb052ad9e0f364fcb7bf4ce81446383"
                    ]
                ],
                "regions_bed_csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.regions.bed.gz.csi:md5,4c72a014f2221c50da97aba3d84f5f5e"
                    ]
                ],
                "quantized_bed_csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.quantized.bed.gz.csi:md5,7b5b75c1fa0459f773cabef50cbe98bd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,04930bdfa05afc685f26d9c56b48be76"
                ],
                "per_base_bed_csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.per-base.bed.gz.csi:md5,06da1743531b338fbcde8c15c1730568"
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.mosdepth.region.dist.txt:md5,1997315b953053f08028e1fe995c2cbc",
                            "test.mosdepth.global.dist.txt:md5,e045f9127572fe2956224545ef777dbf",
                            "test.mosdepth.summary.txt:md5,0086b3d60006c6f009a409080f6fd94c"
                        ]
                    ]
                ],
                "regions_bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.regions.bed.gz:md5,99690daf36751255a87f7b830c24c20f"
                    ]
                ],
                "quantized_bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.quantized.bed.gz:md5,2e6b746ac2fa0c9d9baad787ef914dd0"
                    ]
                ]
            }
        ],
        "timestamp": "2022-12-13T14:01:13+0000"
    },
    "COVERAGE - No target interval file": {
        "content": [
            {
                "per_base_bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.per-base.bed.gz:md5,3bb052ad9e0f364fcb7bf4ce81446383"
                    ]
                ],
                "regions_bed_csi": [
                    
                ],
                "quantized_bed_csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.quantized.bed.gz.csi:md5,7b5b75c1fa0459f773cabef50cbe98bd"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,04930bdfa05afc685f26d9c56b48be76"
                ],
                "per_base_bed_csi": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.per-base.bed.gz.csi:md5,06da1743531b338fbcde8c15c1730568"
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.mosdepth.global.dist.txt:md5,e045f9127572fe2956224545ef777dbf",
                            "test.mosdepth.summary.txt:md5,03d23a4f6c3a0c19a41ca406f7d667d4"
                        ]
                    ]
                ],
                "regions_bed": [
                    
                ],
                "quantized_bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.quantized.bed.gz:md5,2e6b746ac2fa0c9d9baad787ef914dd0"
                    ]
                ]
            }
        ],
        "timestamp": "2022-12-13T14:01:13+0000"
    }
}